lib/rubocop/cop/style/commented_keyword.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for comments put on the same line as some keywords.
+      # These keywords are: `begin`, `class`, `def`, `end`, `module`.
+      #
+      # Note that comments after one-line statements will not be checked. Also,
+      # some comments (such as `:nodoc:` and `rubocop:disable`) are allowed.
+      #
+      # @example
+      #   # bad
+      #   if condition
+      #     statement
+      #   end # end if
+      #
+      #   # bad
+      #   class X # comment
+      #     statement
+      #   end
+      #
+      #   # good
+      #   if condition
+      #     statement
+      #   end
+      #
+      #   # good
+      #   def x; end # comment
+      #
+      #   # good
+      #   class x # :nodoc:
+      #     y
+      #   end
+      class CommentedKeyword < Cop
+        MSG = 'Do not add comments on the same line as the ' \
+              '`%s` keyword.'.freeze
+
+        def investigate(processed_source)
+          heredoc_lines = extract_heredoc_lines(processed_source.ast)
+
+          processed_source.lines.each_with_index do |line, index|
+            next if heredoc_lines.any? { |r| r.include?(index + 1) }
+            next unless offensive?(line)
+
+            range = source_range(processed_source.buffer,
+                                 index + 1,
+                                 (line.index('#'))...(line.length))
+
+            add_offense(range, range)
+          end
+        end
+
+        private
+
+        def keywords
+          %w[begin class def end module]
+        end
+
+        def allowed_comments
+          %w[:nodoc: rubocop:disable]
+        end
+
+        def offensive?(line)
+          keywords.any? { |k| line =~ /^\s*#{k}(\s[^\s]+)?\s*#/ } &&
```

Comment:
It’s unclear what `(\s[^\s]+)?` is meant to match here. It doesn’t seem to match a method signature, for example:
```
>> "def foo(a, b) # comment" =~ /^\s*def(\s[^\s]+)?\s*#/
nil
```

Couldn’t the regex from line 71 just be re-used here?

Reply:
Right. The reason I did that was it might be something like `module X` or `class X`. I'll try to fix the regex.